### ISSUE TITLE
Python: 30 duplicate symbols for architecture x86_64

### ISSUE THREAD

**Issue by davidglivar** [libpd/libpd#143]
Possibly related: https://github.com/libpd/libpd/issues/138

On OSX I've been trying to build the python wrapper per the instructions in the [wiki](https://github.com/libpd/libpd/wiki/Python-API), however, I've been running into some errors while trying to run `make` in the python directory.

Here is the output from `make`:

```
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_array.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_bang.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_canvas.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_clone.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_editor.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_graph.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_guiconnect.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_hdial.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_hslider.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_io.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_mycanvas.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_numbox.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_readwrite.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_rtext.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_scalar.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_template.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_text.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_toggle.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_traversal.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_vdial.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_vslider.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_vumeter.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/m_binbuf.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/m_pd.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/s_inter.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/s_loader.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/x_array.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/x_misc.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/x_scalar.o
duplicate symbol _glist_reloadingabstraction in:
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/g_all_guis.o
    build/temp.macosx-10.11-x86_64-2.7/../pure-data/src/x_text.o
ld: 30 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
error: command 'clang' failed with exit status 1
make: *** [build] Error 1
```

It may also be worth mentioning that this does not occur on Ubuntu Trusty.

Is anyone able to duplicate this issue?

libpd 0.9.1

**Comment by danomatika** [comment]
Try running setup.py with the -fcommon- CFLAG which is what fixed #138:

    CFLAGS="-fcommon" python setup.py

**Comment by danomatika** [comment]
The flag fix is not integrated into the python Makefile.
